src/PSADTree/PSADTreeCmdletBase.cs

```diff
@@ -86,6 +101,17 @@ protected void Push(GroupPrincipal? groupPrincipal, TreeGroup treeGroup)
         if (Recursive.IsPresent || treeGroup.Depth <= Depth)
         {
             _stack.Push((groupPrincipal, treeGroup));
+            return;
+        }
+
+        _truncatedOutput = true;
```

Comment:
I believe this makes more sense, can't test it though. Let me know if it isn't working as expected.